src/main/java/com/github/checkstyle/regression/extract/ExtractInfoProcessor.java

```diff
@@ -58,12 +62,45 @@ private ExtractInfoProcessor() {
     }
 
     /**
+     * Gets the module extract info map from the given branch of checkstyle repository.
+     * @param repoPath the path of checkstyle repository
+     * @param branch   the given branch on which to generate the extract info
+     * @return the full qualified name to module extract info map
+     * @throws InjectException failure when making injection
+     */
+    public static Map<String, ModuleExtractInfo> getModuleExtractInfos(
+            String repoPath, String branch) throws InjectException {
+        final Map<String, ModuleExtractInfo> returnValue;
+        final CheckstyleInjector injector = new CheckstyleInjector(repoPath, branch);
+
+        try {
+            try {
+                final File file = injector.generateExtractInfoFile();
+                final Reader reader = new InputStreamReader(
+                        new FileInputStream(file), Charset.forName("UTF-8"));
+                returnValue = getModuleExtractInfosFromReader(reader);
+            }
+            catch (IOException ex) {
+                throw new InjectException("unable to read to info file", ex);
```

Comment:
This message is not very informative of issue.
Expand on what we were trying to do. Since it is the extract file, it should be in message somewhere.